receiver/signalfxreceiver/signalfxv2_to_metricdata.go

```diff
@@ -17,214 +17,190 @@ package signalfxreceiver
 import (
 	"errors"
 	"fmt"
-	"strconv"
 
-	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
-	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.uber.org/zap"
-	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 var (
 	errSFxNilDatum = errors.New("nil datum value for data-point")
 
-	errSFxUnexpectedInt64DatumType   = errors.New("datum value type of int64 is unexpected")
-	errSFxUnexpectedFloat64DatumType = errors.New("datum value type of float64 is unexpected")
-	errSFxUnexpectedStringDatumType  = errors.New("datum value type of string is unexpected")
-	errSFxStringDatumNotNumber       = errors.New("datum string cannot be parsed to a number")
-	errSFxNoDatumValue               = errors.New("no datum value present for data-point")
+	errSFxNoDatumValue = errors.New("no datum value present for data-point")
 )
 
-// signalFxV2ToMetricsData converts SignalFx proto data points to
-// consumerdata.MetricsData. Returning the converted data and the number of
-// dropped time series.
-func signalFxV2ToMetricsData(
+// signalFxV2ToMetrics converts SignalFx proto data points to pdata.Metrics.
+// Returning the converted data and the number of dropped data points.
+func signalFxV2ToMetrics(
 	logger *zap.Logger,
 	sfxDataPoints []*sfxpb.DataPoint,
-) (consumerdata.MetricsData, int) {
+) (pdata.Metrics, int) {
 
 	// TODO: not optimized at all, basically regenerating everything for each
 	// 	data point.
-	numDroppedTimeSeries := 0
-	md := consumerdata.MetricsData{}
-	metrics := make([]*metricspb.Metric, 0, len(sfxDataPoints))
+	numDroppedDataPoints := 0
+	md := pdata.NewMetrics()
+	md.ResourceMetrics().Resize(1)
+	rm := md.ResourceMetrics().At(0)
+	rm.InitEmpty()
+
+	rm.InstrumentationLibraryMetrics().Resize(1)
+	ilm := rm.InstrumentationLibraryMetrics().At(0)
+	ilm.InitEmpty()
+
+	metrics := ilm.Metrics()
+	metrics.Resize(len(sfxDataPoints))
+
+	i := 0
 	for _, sfxDataPoint := range sfxDataPoints {
 		if sfxDataPoint == nil {
 			// TODO: Log or metric for this odd ball?
 			continue
 		}
 
+		m := metrics.At(i)
 		// First check if the type is convertible and the data point is consistent.
-		metricType, err := convertType(sfxDataPoint)
+		err := fillInType(sfxDataPoint, m)
```

Comment:
Nit: is the split of logic to `fillInType`, followed by a switch with `fill*DataPoint` helpful? Would it be simpler to have just `fillInData` that figures out the type and also fills the data point?

Reply:
Not really it would just mean an additional function that does the same thing.